hw/ip/otbn/dv/smoke/smoke_test.s

```diff
@@ -356,12 +367,85 @@ test_fn_2:
   addi x22, x22, 3
   jalr x0, x1, 0
 
+
+# This function dumps both the register files
+# into gpr_state and wdr_state.
+# The registers aren't clobbered by this function.
+reg_dump:
+  # Dump all the GPRs into gpr_state
+  la x1, gpr_state # (using the x1 to hold a temporary value)
+  sw x2, 0(x1)
+
+  la x2, gpr_state
+  sw  x3,   4(x2) #  1 * 4
+  sw  x4,   8(x2) #  2 * 4
+  sw  x5,  12(x2) #  3 * 4
+  sw  x6,  16(x2) #  4 * 4
+  sw  x7,  20(x2) #  5 * 4
+  sw  x8,  24(x2) #  6 * 4
+  sw  x9,  28(x2) #  7 * 4
+  sw x10,  32(x2) #  8 * 4
+  sw x11,  36(x2) #  9 * 4
+  sw x12,  40(x2) # 10 * 4
+  sw x13,  44(x2) # 11 * 4
+  sw x14,  48(x2) # 12 * 4
+  sw x15,  52(x2) # 13 * 4
+  sw x16,  56(x2) # 14 * 4
+  sw x17,  60(x2) # 15 * 4
+  sw x18,  64(x2) # 16 * 4
+  sw x19,  68(x2) # 17 * 4
+  sw x20,  72(x2) # 18 * 4
+  sw x21,  76(x2) # 19 * 4
+  sw x22,  80(x2) # 20 * 4
+  sw x23,  84(x2) # 21 * 4
+  sw x24,  88(x2) # 22 * 4
+  sw x25,  92(x2) # 23 * 4
+  sw x26,  96(x2) # 24 * 4
+  sw x27, 100(x2) # 25 * 4
+  sw x28, 104(x2) # 26 * 4
+  sw x29, 108(x2) # 27 * 4
+  sw x30, 112(x2) # 28 * 4
+  sw x31, 116(x2) # 29 * 4
+
+  # Dump all the WDRs into wdr_state
+  li x2, 0
+  la x3, wdr_state
+  li x1, 32 # (using the x1 to hold a temporary value)
+  loop x1, 2
+  bn.sid x2++, 0(x3)
+  addi x3, x3, 32
+
+  # Restore the value of x2 and x3 to the value they had
+  # before this function.
+  la x2, gpr_state
+  lw x3, 4(x2)
+  lw x2, 0(x2)
+
+  ret
+
 .section .data
-.word 0x1234abcd
-.word 0xbaadf00d
-.word 0xcafed00d
-.word 0xdeadbeef
-.word 0xfacefeed
-.word 0xaaaaaaaa
-.word 0xbbbbbbbb
-.word 0xcccccccc
+  .word 0x1234abcd
+  .word 0xbaadf00d
+  .word 0xcafed00d
+  .word 0xdeadbeef
+  .word 0xfacefeed
+  .word 0xaaaaaaaa
+  .word 0xbbbbbbbb
+  .word 0xcccccccc
+
+.balign 32
+rnd_value:
+  .rept 4
+  .word 0x99999999
+  .word 0xaaaaaaaa
+  .endr
+
+.global gpr_state
+.balign 32
+gpr_state:
+  .zero (32 / 4) * 30 # not including x0 and x1
+
+.global wdr_state
+.balign 32
+wdr_state:
+  .zero (256 / 4) * 32
```

Comment:
Can this stuff go in an `ifdef deterministic` block? I'm just thinking that a `rnd_value` in the image is going to be a bit confusing if we are actually using RND! :-)